controllers/actions.github.com/autoscalingrunnerset_controller_test.go

```diff
@@ -1465,3 +1551,76 @@ var _ = Describe("Test external permissions cleanup", func() {
 		).Should(BeTrue(), "Expected role to be cleaned up")
 	})
 })
+
+var _ = Describe("Version mismatch cleanup", func() {
+	It("Should delete autoscaling runner set on a version mismatch", func() {
+		ctx := context.Background()
+		autoscalingNS, mgr := createNamespace(GinkgoT(), k8sClient)
+
+		configSecret := createDefaultSecret(GinkgoT(), k8sClient, autoscalingNS.Name)
+
+		controller := &AutoscalingRunnerSetReconciler{
+			Client:                             mgr.GetClient(),
+			Scheme:                             mgr.GetScheme(),
+			Log:                                logf.Log,
+			ControllerNamespace:                autoscalingNS.Name,
+			DefaultRunnerScaleSetListenerImage: "ghcr.io/actions/arc",
+			ActionsClient:                      fake.NewMultiClient(),
+		}
+		err := controller.SetupWithManager(mgr)
+		Expect(err).NotTo(HaveOccurred(), "failed to setup controller")
+
+		originalVersion := build.Version
```

Comment:
Why are we introducing a new pattern here while everywhere else we're using beforeAll / afterAll?